modules/docker/docker_compose.go

```diff
@@ -24,24 +26,24 @@ func RunDockerCompose(t testing.TestingT, options *Options, args ...string) stri
 	return out
 }
 
-// RunDockerComposeAndGetStdout runs docker-compose with the given arguments and options and returns only stdout.
+// RunDockerComposeAndGetStdout runs docker compose with the given arguments and options and returns only stdout.
 func RunDockerComposeAndGetStdOut(t testing.TestingT, options *Options, args ...string) string {
 	out, err := runDockerComposeE(t, true, options, args...)
 	require.NoError(t, err)
 	return out
 }
 
-// RunDockerComposeE runs docker-compose with the given arguments and options and return stdout/stderr.
+// RunDockerComposeE runs docker compose with the given arguments and options and return stdout/stderr.
 func RunDockerComposeE(t testing.TestingT, options *Options, args ...string) (string, error) {
 	return runDockerComposeE(t, false, options, args...)
 }
 
 func runDockerComposeE(t testing.TestingT, stdout bool, options *Options, args ...string) (string, error) {
 	cmd := shell.Command{
-		Command: "docker-compose",
```

Comment:
This will break backward compatibility and will make unusable "old" environments that have only "docker-compose" command.

Was considered to add to "Options" struct feature flags to allow fallback to usage of "docker-compose" ?

Reply:
Could you provide some guidance on how you think this should be implemented? @denis256

Reply:
This is of course your decision, but I would not recommend trying to provide backward compatibility for docker-compose users. Here is why:
1. although both v1 and v2 have nearly identical API, some of the command --flags were not ported from v1 to v2, or ported partly. See: https://docs.docker.com/compose/cli-command-compatibility/.
2. docker-compose v1 will only receive security updates but no new features. I expect docker to sunset v1 soon
3. maintaining compatibility between v1 and v2 will soon become quite time-consuming as v2 will start accumulating new features not present in v1. The maintenance burden will be on your side as docker is not planning to continue developing v1

I recommend to consider this MR a "breaking change" and require users of docker-compose functionality to upgrade their installations. We could add migrations notes, which should mostly point to https://docs.docker.com/compose/cli-command-compatibility/ and installation instructions for v2.

But this is something you as maintainers should decide on your own based on the number of your active users of compose-based tests.

Reply:
Hi,
since `docker compose` was introduced recently, I would say to keep backward compatibility as much as possible maybe by checking if `docker compose version` is available or comparing docker version 

References: https://www.docker.com/blog/announcing-compose-v2-general-availability/

Reply:
Hi, 
an error in case of `docker compose` may appear because of multiple reasons not because compose plugin is not installed...

Can be invoked for example `docker compose version` and checked status code to get right status about plugin installation

Reply:
Thanks for the feedback! @denis256 By status code do you mean this after running `docker compose version`?
```
Docker Compose version v2.5.1
```

Reply:
Would this be better? 
```golang
func runDockerComposeE(t testing.TestingT, stdout bool, options *Options, args ...string) (string, error) {
	cmd := shell.Command{
		Command: "docker",
		// We append --project-name to ensure containers from multiple different tests using Docker Compose don't end
		// up in the same project and end up conflicting with each other.
		Args:       append([]string{"compose", "--project-name", strings.ToLower(t.Name())}, args...),
		WorkingDir: options.WorkingDir,
		Env:        options.EnvVars,
		Logger:     options.Logger,
	}

	dockerComposeVersionCmd := icmd.Command("docker compose version")

	result := icmd.RunCmd(dockerComposeVersionCmd)

	if result.ExitCode != 0 {
		cmd = shell.Command{
			Command:    "docker-compose",
			Args:       append([]string{"--project-name", strings.ToLower(t.Name())}, args...),
			WorkingDir: options.WorkingDir,
			Env:        options.EnvVars,
			Logger:     options.Logger,
		}
	}

	if stdout {
		return shell.RunCommandAndGetStdOut(t, cmd), nil
	}
	return shell.RunCommandAndGetOutputE(t, cmd)
}
```

Reply:
Hi,
I think this approach is better, not sure why should be created each time `cmd` struct with `docker-compose`, maybe first run `docker compose version` and then create respective structure

Reply:
Thanks for the feedback, would we not need to have two commands since there are two conditions?
1. `docker compose version` command fails so use `docker-compose` command
2. `docker compose version` succeeds, so use `docker` command with `compose` as an argument

Reply:
And `cmd` needs to be initialised outside of the `if` block, otherwise the following error is returned:
```
cmd declared but not used compiler(UnusedVar)
```

Reply:
Is this better?
```golang
func runDockerComposeE(t testing.TestingT, stdout bool, options *Options, args ...string) (string, error) {
	var cmd shell.Command

	dockerComposeVersionCmd := icmd.Command("docker", "compose", "version")

	if icmd.RunCmd(dockerComposeVersionCmd).ExitCode != 0 {
		cmd = shell.Command{
			Command: "docker",
			// We append --project-name to ensure containers from multiple different tests using Docker Compose don't end
			// up in the same project and end up conflicting with each other.
			Args:       append([]string{"compose", "--project-name", strings.ToLower(t.Name())}, args...),
			WorkingDir: options.WorkingDir,
			Env:        options.EnvVars,
			Logger:     options.Logger,
		}
	}
	cmd = shell.Command{
		Command:    "docker-compose",
		Args:       append([]string{"--project-name", strings.ToLower(t.Name())}, args...),
		WorkingDir: options.WorkingDir,
		Env:        options.EnvVars,
		Logger:     options.Logger,
	}

	if stdout {
		return shell.RunCommandAndGetStdOut(t, cmd), nil
	}
	return shell.RunCommandAndGetOutputE(t, cmd)
}

```

Reply:
Is this okay to go ahead? @denis256